### PR TITLE
Fix build for 1.5.4

### LINKED ISSUE #297: `papermill` not compatible with `py>=312`
***xref:*** https://github.com/conda-forge/dagster-feedstock/pull/296#issuecomment-1774239997

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

----

Resolves #297

### CODE REVIEW

**Comment by dhirschfeld** [comment]
@conda-forge-admin, please rerender

**Comment by dhirschfeld** [comment]
This pin should allow `dagster` to be built and installed on all platforms.

You should be able to install it in a Python 3.12 environment and should be fine there as long as you don't want to use `dagstermill`. I think that's the best we can do since upstream `papermill` isn't compatible with Python 3.12; though it should still work if someone installs `papermill` from git.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1826,7 +1826,8 @@ outputs:
 
     test:
       requires:
-        - python >=3.8
+        # FIXME: https://github.com/conda-forge/dagster-feedstock/issues/297
```
Comment (bollwyvl) [inline recipe/meta.yaml:1829]: This is fine, but should also be fixed at the source with a PR @conda-forge/papermill, and likely a PR to `repodata-patches` to `tighten_depends` on `papermill` vs `python`.

Comment (dhirschfeld) [reply recipe/meta.yaml:1829]: I'll see if I can tick those off this week...

Comment (dhirschfeld) [reply recipe/meta.yaml:1829]: Part 1 - https://github.com/conda-forge/papermill-feedstock/pull/40

Comment (bollwyvl) [reply recipe/meta.yaml:1829]: Sry, was looking a tad deeper: really just needs to be fixed at the source, so it this case ansiwrap:

- https://github.com/conda-forge/ansiwrap-feedstock/pull/3
- https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/577

Comment (dhirschfeld) [reply recipe/meta.yaml:1829]: Right, yes - that's better. It will work for anything depending on `ansiwrap` rather than just fixing `papermill`.

**Pull request merged**